sktime/classification/shapelet_based/_stc.py

```diff
@@ -84,11 +79,26 @@ def fit(self, X, y):
         -------
         self : object
          """
-        X, y = check_X_y(X, y, enforce_univariate=True)
-        self.n_classes = np.unique(y).shape[0]
-        self.classes_ = class_distribution(np.asarray(y).reshape(-1, 1))[0][0]
+        _X, _y = check_X_y(X, y, enforce_univariate=True)
```

Comment:
Why do you create the underscore objects? Should work without the underscores, no?

Reply:
created the underscores as that was a suggestion you left me before about not overwriting input args
I'm happy to remove that.

Reply:
actually there was also a weird error, i think with the fact X was getting shuffled so i was getting an idemptoency failure because the shapes were mismatched.

Reply:
ah sorry I meant not overwriting hyper-parameters set in the constructor, input args can be overwritten